sound/soc/sof/sof-audio.c

```diff
@@ -92,14 +101,14 @@ int sof_widget_free(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 		.id = swidget->comp_id,
 	};
 	struct sof_ipc_reply reply;
-	int ret;
+	int ret = 0;
 
 	if (!swidget->private)
 		return 0;
 
-	/* only free when refcount is 0 */
-	if (--swidget->use_count)
-		return 0;
+	/* only free when use_count is 0 */
+	if (--swidget->use_count > 0)
+		return ret;
```

Comment:
I'd keep `return 0;` here and remove the otherwise redundant `ret` initialisation. Especially since there's already a `return 0;` above, so, having `return ret;` here makes it look like this really can be anything different from `0`